app/controllers/microposts_controller.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+class MicropostsController < ApplicationController
+  before_action :set_micropost, only: %i[show edit update destroy]
+
+  def index
+    @microposts = Micropost.all.order("created_at desc")
+  end
+
+  def show; end
+
+  def new
+    @micropost = Micropost.new
+  end
+
+  def edit; end
+
+  def create
+    @micropost = Micropost.new(micropost_params)
+    if @micropost.save
+      redirect_to @micropost, notice: '新規投稿を作成しました。'
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    if @micropost.update(micropost_params)
+      redirect_to @micropost, notice: '投稿を更新しました。', status: :see_other
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @micropost.destroy
```

Comment:
ここで destroy がエラーになった場合にクライアント側は気づけなくなるので `destroy!` を使いましょう。
`!` の有り無しの挙動の違いはあとで見といてくださいー
```suggestion
    @micropost.destroy!
```